### PR TITLE
[ENH] Add reduce on plateau learning rate decay for FCN ResNet and MLP deep classifiers

### LINKED ISSUE #320: [ENH] add ReduceOnPlateau callbacks for the learning rate to all deep learning classifiers
as a default callback it should be added to not overfit the models, when getting in InceptionTime before there was an issue with this callbacks during test checks. To see if it works now.

- [x] InceptionTime
- [x] FCN
- [x] ResNet
- [x] MLP

The other classifiers do not use them by default, for more info check [dl4tsc](https://github.com/hfawaz/dl-4-tsc/tree/master/classifiers)

### PR DESCRIPTION
**Pull request by hadifawaz1999**
Resolve #320

### CODE REVIEW

**Comment by TonyBagnall** [comment]
fantastic, thanks @hadifawaz1999

**Review by TonyBagnall — APPROVED** [review]
